File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        ans=0

        if len(chars)==1:
            return len(chars)

        count=1
        for i in range(1, len(chars)):
            if chars[i]==chars[i-1]:
                count+=1
            else:
                chars[ans]=chars[i-1]
                ans+=1
                if count>1:
                    tmp=str(count)
                    for j in tmp:
                        chars[ans]=j
                        ans+=1
                count=1

        chars[ans]=chars[-1]
        ans+=1
        if count>1:
            tmp=str(count)
            for j in tmp:
                chars[ans]=j
                ans+=1

        return ans

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int count=1, ans=0;
        for (int i=1; i<chars.length; i++) {
            if (chars[i]==chars[i-1]) {
                count++;
                continue;
            }
            else {
                chars[ans++]=chars[i-1];
                if (count>1) {
                    String tmp=Integer.toString(count);
                    for (int j=0; j<tmp.length(); j++) {
                        chars[ans++]=tmp.charAt(j);
                    }
                }
                count=1;
            }
        }

        chars[ans++]=chars[chars.length-1];
        if (count>1) {
            String tmp=Integer.toString(count);
            for (int j=0; j<tmp.length(); j++) {
                chars[ans++]=tmp.charAt(j);
            }
        }

        return ans;
    }
}